### DISCUSSION TITLE
How to only allow ['internal'] transport during Registration for a new User?

### DISCUSSION THREAD

**Discussion by h1zqeel** [MasterKale/SimpleWebAuthn#217] [General]
how can implement something like webauthn.io where on register we see a pop up for fingerprint/faceid directly ?

**Comment by MasterKale** [comment]
For that you'll want to take advantage of the [`authenticatorSelection` argument](https://www.w3.org/TR/webauthn-2/#dictionary-makecredentialoptions) in the options you pass to `navigator.credentials.create()`. Setting `authenticatorAttachment` to `"platform"` should do the trick:

```js
navigator.credentials.create({
  // ...
  publicKey: {
    authenticatorSelection: {
      authenticatorAttachment: 'platform'
    },
  },
});
```

**Reply by h1zqeel** [reply]
thanks alot for the quick response, it would be nice if you guys could document this and all the other possibilities on your website :)

**Comment by DanielRios549** [comment]
How to pass this information using this lib?

**Reply by MasterKale** [reply]
Great question! You can specify `"platform"` authenticator attachment like this when calling `generateRegistrationOptions()`:

```ts
generateRegistrationOptions({
  // ...
  authenticatorSelection: {
    authenticatorAttachment: 'platform',
  },
});
```
